src/psyclone/psyir/symbols/symbol_table.py

```diff
@@ -1208,38 +1208,29 @@ def _validate_remove_routinesymbol(self, symbol):
                 pass
 
         # Check for any references to it.
-        # pylint: disable=import-outside-toplevel
-        from psyclone.core import Signature, VariablesAccessMap
-        vam = VariablesAccessMap()
+        symbols = self.get_all_accessed_symbols()
+        location = "the provided symbol_table"
         if self.node:
-            vam.update(self.node.reference_accesses())
-        vam.update(self.reference_accesses())
-        sig = Signature(symbol.name)
-        if sig not in vam:
-            return
-
-        # TODO #2424 - ideally AccessSequence.AccessInfo or
-        # Signature would store the actual Symbol that the access is to. In
-        # the absence of that, we have to examine each access to determine
-        # the Symbol.
-        from psyclone.psyir.nodes.reference import Reference
+            symbols.update(self.node.get_all_accessed_symbols())
+            location = f"the '{self.node.name}' scope"
+        # pylint: disable=import-outside-toplevel
         from psyclone.psyir.symbols.generic_interface_symbol import (
             GenericInterfaceSymbol)
-        try:
-            for access in vam[sig]:
-                if isinstance(access.node, GenericInterfaceSymbol):
-                    for rinfo in access.node.routines:
-                        if rinfo.symbol is symbol:
-                            raise ValueError()
-                else:
-                    for ref in access.node.walk(Reference):
-                        if ref.symbol is symbol:
-                            raise ValueError()
-        except ValueError:
-            # pylint: disable-next=raise-missing-from
-            raise ValueError(
-                f"Cannot remove RoutineSymbol '{symbol.name}' because it is "
-                f"referenced by {access.description}")
+        for sym in symbols:
+            if isinstance(sym, GenericInterfaceSymbol):
+                for rinfo in sym.routines:
```

Comment:
Do we need this? I thought that GenericInterfaceSymbol.get_all_accessed_symbols() will return all of its constituent RoutineSymbols so we should only need `if symbol not in symbols` now?